In [3]:
import os
from tabulate import tabulate

In [4]:
# path
file_path = 'Model/Scenario1/'

# loop file path
for i in range(4):
    print(i+1)
    file_path = 'Model/Scenario' + str(i+1) + '/'
    print(file_path)

1
Model/Scenario1/
2
Model/Scenario2/
3
Model/Scenario3/
4
Model/Scenario4/


tabel kategori tindakan dengan rata-rata durasi video tindakan

gambar perbandingan cuplikan video acak dari masing-masing kategori tindakan

extraction time

In [6]:
# Time Conversion Function
def convert(seconds):
    seconds = seconds % (24 * 3600)
    hour = seconds // 3600
    seconds %= 3600
    minutes = seconds // 60
    seconds %= 60
 
    return "%d:%02d:%02d" % (hour, minutes, seconds)

In [15]:

import time
# start time
start = time.time()

# loop with 3s delay
for i in range(4):
    print(i+1)
    file_path = 'Model/Scenario' + str(i+1) + '/'
    print(file_path)
    # loop with 3s delay
    time.sleep(3)

# end time
end = time.time()
end_time = end - start
print(end_time)

1
Model/Scenario1/
2
Model/Scenario2/
3
Model/Scenario3/
4
Model/Scenario4/
12.019589185714722


In [31]:
with open('Model/Scenario1/extraction_execution_time.txt', 'r') as file:
    data = file.read().replace('\n', '')
    print(data)
    print(convert(float(data)))

10267.57
2:51:07


In [56]:
execution_time = []

for i in range(4):
    with open(os.path.join('Model/Scenario' + str(i+1) + '/', 'extraction_execution_time.txt'), 'r') as f:
        time=f.read()
        execution_time.append([
            f"Scenario {str(i+1)}", 
            time, 
            convert(float(time))
            ])

print(tabulate(execution_time, headers=['Execution', 'Seconds', 'Hours'], tablefmt='pretty'))

+------------+----------+---------+
| Execution  | Seconds  |  Hours  |
+------------+----------+---------+
| Scenario 1 | 10267.57 | 2:51:07 |
| Scenario 2 |  149.11  | 0:02:29 |
| Scenario 3 | 10267.57 | 2:51:07 |
| Scenario 4 |  149.11  | 0:02:29 |
+------------+----------+---------+


In [68]:
# highlight first row in table tabulate
print(tabulate(execution_time, headers=['Execution', 'Seconds', 'Hours'], tablefmt='html'))

<table>
<thead>
<tr><th>Execution  </th><th style="text-align: right;">  Seconds</th><th>Hours  </th></tr>
</thead>
<tbody>
<tr><td>Scenario 1 </td><td style="text-align: right;"> 10267.6 </td><td>2:51:07</td></tr>
<tr><td>Scenario 2 </td><td style="text-align: right;">   149.11</td><td>0:02:29</td></tr>
<tr><td>Scenario 3 </td><td style="text-align: right;"> 10267.6 </td><td>2:51:07</td></tr>
<tr><td>Scenario 4 </td><td style="text-align: right;">   149.11</td><td>0:02:29</td></tr>
</tbody>
</table>


<table>
<thead>
<tr><th>Execution  </th><th style="text-align: right;">  Seconds</th><th>Hours  </th></tr>
</thead>
<tbody>
<tr bgcolor="green"><td>Scenario 1 </td><td style="text-align: right;"> 10267.6 </td><td>2:51:07</td></tr>
<tr><td>Scenario 2 </td><td style="text-align: right;">   149.11</td><td>0:02:29</td></tr>
<tr><td>Scenario 3 </td><td style="text-align: right;"> 10267.6 </td><td>2:51:07</td></tr>
<tr><td>Scenario 4 </td><td style="text-align: right;">   149.11</td><td>0:02:29</td></tr>
</tbody>
</table>


total data extracted

grid search results

plot model

plot model loss and accuracy

evaluate model

evaluation predicted matrix

confusion matrix